segmentstore/storage/impl/src/main/java/io/pravega/segmentstore/storage/impl/filesystem/FileSystemStorage.java

```diff
@@ -409,7 +411,9 @@ private Throwable translateException(String segmentName, Throwable e) {
             retVal = new IllegalArgumentException(e.getMessage());
         }
 
-        if (e instanceof AccessDeniedException) {
+        if (e instanceof AccessControlException
+                || e instanceof AccessDeniedException
```

Comment:
One more point. In the S3 case, we are letting the `AccessDeniedException` to fall through, but not for file system. Is this the correct behavior?